Compiler warnings in WCSimEventAction.{hh,cc}

Compiling WCSim with gcc/g++ version 4.8.5 reports the warnings related to WCSimEventAction.{hh,cc} in the attached file [make.log](https://github.com/spradlin/WCSim/files/6566356/make.log).
The warnings were generated by `touch`ing the source files WCSimEventAction.{hh,cc} and `make`ing.

See the code at https://github.com/spradlin/WCSim/blob/e5d51e71612d6a9fdb69a425518a519ef8450aad/src/WCSimEventAction.cc#L207-L215 for context of the warning
```
src/WCSimEventAction.cc: In member function 'virtual void WCSimEventAction::EndOfEventAction(const G4Event*)':
src/WCSimEventAction.cc:211:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   for( Int_t u=0; u<nvtxs; u++ ){
                     ^
```
It and other identical warnings occur becuase the loop index is signed rather than unsigned.

The return type of `WCSimPrimaryGeneratorAction::GetNvtxs()` is `G4int`, which is an alias of `int` in the GEANT4 header `G4Types.hh`.  There is no reason for the variable `nvtxs` to be unsigned.  Probably the best solution is to make both the variable `ntvxs` and the loop indexes `G4int`s.

Well, the best solution would probably be to use `auto` variables for guaranteed and durable consistency, but that would also obscure the type.  If the signed/unsigned distinction is important to the code's function, it could get lost.

The warning
```
src/WCSimEventAction.cc: In member function 'void WCSimEventAction::FillRootEvent(G4int, const ntupleStruct&, G4TrajectoryContainer*, WCSimWCDigitsCollection*, WCSimWCTriggeredDigitsCollection*, G4String)':
src/WCSimEventAction.cc:810:62: warning: declaration of 'jhfNtuple' shadows a global declaration [-Wshadow]
                                      G4String detectorElement)
                                                              ^
In file included from include/WCSimRootEvent.hh:18:0,
                 from include/WCSimRunAction.hh:11,
                 from src/WCSimEventAction.cc:3:
include/jhfNtuple.h:55:28: warning: shadowed declaration is here [-Wshadow]
 extern struct ntupleStruct jhfNtuple;
                            ^
```
is interesting.

The only calls to `WCSimEventAction::FillRootEvent()` that i have so far found are in `WCSimEventAction::EndOfEventAction()`:
https://github.com/spradlin/WCSim/blob/e5d51e71612d6a9fdb69a425518a519ef8450aad/src/WCSimEventAction.cc#L658-L672

In each of these cases, the global instance of `jhfNtuple` is passed into `WCSimEventAction::FillRootEvent()` as the argument with a same local name.

The most transparent solution is probably to change the name of the local variable.

--- a/src/WCSimEventAction.cc
+++ b/src/WCSimEventAction.cc
@@ -172,10 +172,6 @@ void WCSimEventAction::BeginOfEventAction(const G4Event*)
 {
   if(!ConstructedDAQClasses) {
     CreateDAQInstances();
-
-    //and save options in output file
-    G4DigiManager* DMman = G4DigiManager::GetDMpointer();
-
   }
 }
 
@@ -204,11 +200,11 @@ void WCSimEventAction::EndOfEventAction(const G4Event* evt)
   G4int         event_id = evt->GetEventID();
 //  G4int         mode     = generatorAction->GetMode();
 
-  unsigned int      nvtxs   = generatorAction->GetNvtxs();
+  G4int      nvtxs   = generatorAction->GetNvtxs();
   G4ThreeVector vtxs[MAX_N_VERTICES];
   G4int         vtxsvol[MAX_N_VERTICES];
   G4double      vtxTimes[MAX_N_VERTICES];
-  for( Int_t u=0; u<nvtxs; u++ ){
+  for( G4int u=0; u<nvtxs; u++ ){
     vtxs[u]      = generatorAction->GetVtx(u);
     vtxsvol[u]   = WCSimEventFindStartingVolume(vtxs[u]);
     vtxTimes[u]  = generatorAction->GetVertexTime(u);
@@ -409,10 +405,10 @@ void WCSimEventAction::EndOfEventAction(const G4Event* evt)
   // ----------------------------------------------------------------------
   //  Fill Ntuple
   // ----------------------------------------------------------------------
-   for(unsigned int ivx = 0;ivx<nvtxs;ivx++)
+   for(G4int ivx = 0;ivx<nvtxs;ivx++)
      jhfNtuple.mode[ivx]   = generatorAction->GetMode(ivx);         // interaction mode
    jhfNtuple.nvtxs = nvtxs;       // number of vertices
-   for( Int_t u=0; u<nvtxs; u++ ){
+   for( G4int u=0; u<nvtxs; u++ ){
      jhfNtuple.vtxsvol[u] = vtxsvol[u];       // volume of vertex
      // unit mismatch between geant4 and reconstruction, M Fechner
      jhfNtuple.vtxs[u][0] = vtxs[u][0]/cm; // interaction vertex
@@ -431,7 +427,7 @@ void WCSimEventAction::EndOfEventAction(const G4Event* evt)
 
    // First two tracks are special: beam and target
 
-   for( Int_t u=0; u<nvtxs; u++ ){
+   for( G4int u=0; u<nvtxs; u++ ){
      /////////////////////////////////
      // npar = 0        NEUTRINO /////
      /////////////////////////////////
@@ -525,7 +521,7 @@ void WCSimEventAction::EndOfEventAction(const G4Event* evt)
 
   // Draw Charged Tracks
   G4int PDG_e=11,PDG_v_e=12,PDG_gam=22;
-  for( Int_t u=0; u<nvtxs; u++ ){
+  for( G4int u=0; u<nvtxs; u++ ){
     G4int trkid_e=INT_MAX,trkid_v_e=INT_MAX,trkid_gam=INT_MAX;
     G4int idx_e=INT_MAX,idx_v_e=INT_MAX,idx_gam=INT_MAX;
     for (G4int i=0; i < n_trajectories; i++) 
@@ -803,7 +799,7 @@ G4int WCSimEventAction::WCSimEventFindStoppingVolume(G4String stopVolumeName)
 }
 
 void WCSimEventAction::FillRootEvent(G4int event_id, 
-                                     const struct ntupleStruct& jhfNtuple,
+                                     const struct ntupleStruct& myNtuple,
                                      G4TrajectoryContainer* TC,
                                      WCSimWCDigitsCollection* WCDC_hits,
                                      WCSimWCTriggeredDigitsCollection* WCDC,
@@ -834,7 +830,7 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
 	wcsimrootevent = wcsimrootsuperevent->GetTrigger(index);
 	wcsimrootevent->SetHeader(event_id,0,
 				   0,index+1); // date & # of subevent 
-	wcsimrootevent->SetMode(jhfNtuple.mode[0]);
+	wcsimrootevent->SetMode(myNtuple.mode[0]);
       }
       wcsimrootevent->SetTriggerInfo(WCTM->GetTriggerType(index),
 				     WCTM->GetTriggerInfo(index));
@@ -848,27 +844,27 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
 
   // Fill other info for this event
 
-  wcsimrootevent->SetMode(jhfNtuple.mode[0]);
-  wcsimrootevent->SetNvtxs(jhfNtuple.nvtxs);
-  for( Int_t u=0; u<jhfNtuple.nvtxs; u++ ){
-    wcsimrootevent->SetVtxsvol(u,jhfNtuple.vtxsvol[u]);
+  wcsimrootevent->SetMode(myNtuple.mode[0]);
+  wcsimrootevent->SetNvtxs(myNtuple.nvtxs);
+  for( Int_t u=0; u<myNtuple.nvtxs; u++ ){
+    wcsimrootevent->SetVtxsvol(u,myNtuple.vtxsvol[u]);
     for (int j=0;j<4;j++)
       {
-       wcsimrootevent->SetVtxs(u,j,jhfNtuple.vtxs[u][j]);
+       wcsimrootevent->SetVtxs(u,j,myNtuple.vtxs[u][j]);
       }
-      wcsimrootevent->SetMode(u,jhfNtuple.mode[u]);
+      wcsimrootevent->SetMode(u,myNtuple.mode[u]);
   }
-  wcsimrootevent->SetJmu(jhfNtuple.jmu);
-  wcsimrootevent->SetJp(jhfNtuple.jp);
-  wcsimrootevent->SetNpar(jhfNtuple.npar);
-  wcsimrootevent->SetVecRecNumber(jhfNtuple.vecRecNumber);
+  wcsimrootevent->SetJmu(myNtuple.jmu);
+  wcsimrootevent->SetJp(myNtuple.jp);
+  wcsimrootevent->SetNpar(myNtuple.npar);
+  wcsimrootevent->SetVecRecNumber(myNtuple.vecRecNumber);
 
   // Add the tracks with the particle information
-  // First two tracks come from jhfNtuple, as they are special
+  // First two tracks come from myNtuple, as they are special
 
   int k;
   //Modify to add decay products
-  for (k=0;k<jhfNtuple.npar;k++) // should be just 2
+  for (k=0;k<myNtuple.npar;k++) // should be just 2
   {
     double dir[3];
     double pdir[3];
@@ -876,27 +872,27 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
     double start[3];
     for (int l=0;l<3;l++)
     {
-      dir[l]=jhfNtuple.dir[k][l];
-      pdir[l]=jhfNtuple.pdir[k][l];
-      stop[l]=jhfNtuple.stop[k][l];
-      start[l]=jhfNtuple.start[k][l];
-      //G4cout<< "start[" << k << "][" << l <<"]: "<< jhfNtuple.start[k][l] <<G4endl;
+      dir[l]=myNtuple.dir[k][l];
+      pdir[l]=myNtuple.pdir[k][l];
+      stop[l]=myNtuple.stop[k][l];
+      start[l]=myNtuple.start[k][l];
+      //G4cout<< "start[" << k << "][" << l <<"]: "<< myNtuple.start[k][l] <<G4endl;
     }
 
     // Add the track to the TClonesArray
-    wcsimrootevent->AddTrack(jhfNtuple.ipnu[k], 
-			      jhfNtuple.flag[k], 
-			      jhfNtuple.m[k], 
-			      jhfNtuple.p[k], 
-			      jhfNtuple.E[k], 
-			      jhfNtuple.startvol[k], 
-			      jhfNtuple.stopvol[k], 
+    wcsimrootevent->AddTrack(myNtuple.ipnu[k], 
+			      myNtuple.flag[k], 
+			      myNtuple.m[k], 
+			      myNtuple.p[k], 
+			      myNtuple.E[k], 
+			      myNtuple.startvol[k], 
+			      myNtuple.stopvol[k], 
 			      dir, 
 			      pdir, 
 			      stop,
 			      start,
-			      jhfNtuple.parent[k],
-			     jhfNtuple.time[k],0); 
+			      myNtuple.parent[k],
+			     myNtuple.time[k],0); 
   }
 
   // the rest of the tracks come from WCSimTrajectory
@@ -1068,7 +1064,7 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
   // Add the Cherenkov hits
   wcsimrootevent = wcsimrootsuperevent->GetTrigger(0);
 
-  wcsimrootevent->SetNumTubesHit(jhfNtuple.numTubesHit);
+  wcsimrootevent->SetNumTubesHit(myNtuple.numTubesHit);
 
 #ifdef _SAVE_RAW_HITS
 
@@ -1083,7 +1079,7 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
     wcsimrootevent->SetNumTubesHit(WCDC_hits->entries());
     std::vector<double> truetime, smeartime;
     std::vector<int>   primaryParentID;
-    double hit_time_smear, hit_time_true;
+    double hit_time_true;
     int hit_parentid;
     //loop over the DigitsCollection
     for(int idigi = 0; idigi < WCDC_hits->entries(); idigi++) {
@@ -1094,7 +1090,7 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
 	truetime.push_back(hit_time_true);
 	primaryParentID.push_back(hit_parentid);
 #ifdef _SAVE_RAW_HITS_VERBOSE
-	hit_time_smear = (*WCDC_hits)[idigi]->GetTime(id);
+        double hit_time_smear = (*WCDC_hits)[idigi]->GetTime(id);
 	smeartime.push_back(hit_time_smear);
 #endif
       }//id
@@ -1215,7 +1211,6 @@ void WCSimEventAction::FillRootEvent(G4int event_id,
   //G4cout <<"WCFV digi sumQ:"<<std::setw(4)<<wcsimrootevent->GetSumQ()<<"  ";
   //  }
   
-  TTree* tree = GetRunAction()->GetTree();
   TBranch* branch = GetRunAction()->GetBranch(detectorElement);
   branch->Fill();
   //tree->Fill();

